### ISSUE TITLE
java.lang.Throwable: Can't remove document listener

### ISSUE THREAD

**Issue by odisseus** [sourcegraph/jetbrains#1608]
### Cody Version

52c7b529d7208f66a124e56352ac16373857c0d2

### IDE Information

IDEA CE 2022.1, IDEA CE 2024.1

### Describe the bug

To reproduce, trigger "Generate unit tests" in such a way that a new file would be created. This happens reliably every time.

The tests are generated, but there is an IDE error popup.

### Expected behavior

No errors

### Additional context

```
java.lang.Throwable: Can't remove document listener (FixupSessionDocumentListener@2cfad81d). Registered listeners: [com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl$1@1aef43e5, 626, 159, Multicaster, 116, 9, 7, 0, 38, 4, 5, 81, 0, 8, 78, 0, 0, [], com.intellij.openapi.editor.impl.view.LogicalPositionCache@719bf2fc, com.intellij.openapi.editor.impl.view.TextLayoutCache@628cd228, com.intellij.openapi.editor.ex.util.LexerEditorHighlighter(com.intellij.lexer.JavaHighlightingLexer): '

// generated Java code

', 
use soft wraps: false, tab width: 4, additional columns: false, update in progress: false, bulk update in progress: false, dirty: true, deferred regions: []
appliance manager state: recalculation in progress: false; event being processed: null, available width: 944, visible width: 0, dirty: true
soft wraps mapping info: Soft wraps affected by current update: []
soft wraps: [], com.intellij.openapi.editor.impl.view.EditorSizeManager@404f08a2, com.intellij.openapi.editor.impl.CaretModelImpl@20f360fe, com.intellij.openapi.editor.impl.InlayModelImpl@3bd23a6f, com.intellij.openapi.editor.impl.EditorImpl$EditorDocumentAdapter@50e7c827, Multicaster, com.intellij.openapi.editor.impl.event.EditorEventMulticasterImpl$1@897fdff, {}, com.intellij.openapi.editor.impl.ScrollingModelImpl$1@23e3f0d8, com.intellij.codeInsight.codeVision.CodeVisionHost$subscribeForFrontendEditor$5@7b24ca1e, com.intellij.openapi.fileEditor.impl.text.TextEditorComponent$MyDocumentListener@156c83ad]
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at com.intellij.openapi.editor.impl.DocumentImpl.removeDocumentListener(DocumentImpl.java:1004)
	at com.sourcegraph.cody.edit.sessions.FixupSession.dispose(FixupSession.kt:413)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:139)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:169)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:219)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:207)
	at com.sourcegraph.cody.edit.sessions.FixupSession.completionFuture$lambda$1(FixupSession.kt:94)
	at com.sourcegraph.cody.vscode.CancellationToken.lambda$onFinished$1(CancellationToken.java:25)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at com.sourcegraph.cody.vscode.CancellationToken.dispose(CancellationToken.java:41)
	at com.sourcegraph.cody.edit.sessions.FixupSession.dismiss(FixupSession.kt:281)
	at com.sourcegraph.cody.agent.CodyAgentService$2.invoke$lambda$4(CodyAgentService.kt:77)
	at com.sourcegraph.cody.agent.CodyAgentClient.lambda$acceptOnEventThread$1(CodyAgentClient.java:143)
	at com.sourcegraph.cody.agent.CodyAgentClient.lambda$acceptOnEventThread$0(CodyAgentClient.java:126)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:215)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:197)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:511)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:69)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:112)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:42)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:803)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

**Comment by odisseus** [comment]
Apparently this has been caused by #1603.

**Comment by odisseus** [comment]
This seems to be fixed, possibly by #1621.
